pkgs/os-specific/linux/dpdk/default.nix

```diff
@@ -59,30 +65,28 @@ in stdenv.mkDerivation rec {
   ++ lib.optional (!shared) "-Ddefault_library=static"
   ++ lib.optional stdenv.isx86_64 "-Dmachine=nehalem"
   ++ lib.optional stdenv.isAarch64 "-Dmachine=generic"
-  ++ lib.optional mod "-Dkernel_dir=${placeholder "kmod"}/lib/modules/${kernel.modDirVersion}"
+  ++ lib.optionals mod [
+    "-Dkernel_dir=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "-Dkernel_install_dir=${placeholder "kmod"}/lib/modules/${kernel.modDirVersion}"
```

Comment:
I am getting this error in basically all kernel module builds:

```
error: builder for '/nix/store/87vb4bahw9fpm0mzb796bgp8rnf7z1si-dpdk-21.11-5.4.173.drv' failed with exit code 1;
       last 10 log lines:
       > The Meson build system
       > Version: 0.60.3
       > Source dir: /build/dpdk-21.11
       > Build dir: /build/dpdk-21.11/build
       > Build type: native build
       > Program cat found: YES (/nix/store/vizjhz04x6xl57x2vrpqa52j8q6rkjfh-coreutils-9.0/bin/cat)
       >
       > meson.build:4:0: ERROR: Unknown options: "kernel_install_dir"
       >
       > A full log can be found at /build/dpdk-21.11/build/meson-logs/meson-log.txt
```

Should be reproducible by simply running nixpkgs-review on this pr.

Reply:
Indeed, strange that I didn't have these issues when building. Fixed.